### DISCUSSION TITLE
How to change theme for inputs?

### DISCUSSION THREAD

**Discussion by DaviCaamano** [chakra-ui/chakra-ui#5976] [Q&A]
I'd like to create some default variants for input text boxes and checkboxes.

But when I extend the theme for `Input` it doesn't seem to work.

```
const chakraTheme = extendTheme({
  breakpoints,
  components: {
    Input: {
      variants: {
        text: {
          width: '100%',
          padding: '0.5rem 1.25rem',
          display: 'inline-block',
          border: '1px solid #ccc',
          boxSizing: 'border-box',
          borderRadius: 0,
          my: '0.25rem',
          bg: '#E7F0FD',
        },
        check: {
          w: '1.25rem',
          h: '1.25rem',
          p: 0,
        },
      },
    },
  },
});
```

**Comment by BillyPurvis** [comment]
Hello! 

If I'm reading this correct, you'd like to create a variant called 'text'? If so, you just need to target a specific part of the [multipart component](https://chakra-ui.com/docs/styled-system/theming/component-style#single-part-and-multipart-components). You can find the correct name of the part reading the component's theme definition in the repository. [Here's the Input's theme definition.](https://github.com/chakra-ui/chakra-ui/blob/main/packages/theme/src/components/input.ts). 

If you take a look at the default variants, you can see how to define your own and target the relevant part. If you want the variant you created to be the default variant applied for the Input component, you can use the `defaultProps` property on the component's theme object. 

```
extendTheme({
  components: {
    Input: {
      variants: {
        text: {
          field: {
            width: "100%",
            padding: "0.5rem 1.25rem",
            display: "inline-block",
            border: "1px solid #ccc",
            boxSizing: "border-box",
            borderRadius: 0,
            my: "0.25rem",
            bg: "#E7F0FD"
          },          
        },        
      },
      defaultProps: {
        variant: "text"
      }
    }
  }
});
```

I've provided a CodeSanbox for you, but I'm not sure if you wanted 'check' to be a variant, or target an icon within either the left or right addon / element components. 

**Code Sandbox:** https://codesandbox.io/s/falling-lake-r6f7m4?file=/src/index.tsx

It's worth nothing this [discussion here](https://github.com/chakra-ui/chakra-ui/discussions/2974) which talks about extending a variant within the theme, in-case you'd like to inherit some of the default variants styles.

**Comment by DaviCaamano** [comment]
Thanks!

What are the part names?

I can't find documentation for input parts.

**Reply by BillyPurvis** [reply]
For the `Input` component, there's a few parts; `"addon", "field", "element"`.  You can find them in a component's theme definition file imported from `@chakra-ui/anatomy` package. 

You can see on the [first line in the Input's theme file](https://github.com/chakra-ui/chakra-ui/blob/main/packages/theme/src/components/input.ts#L1) the imported anatomy (and parts) from the anatomy package.  You'll also be able to look at all the other component's parts [here](https://github.com/chakra-ui/chakra-ui/blob/main/packages/anatomy/src/index.ts)
